com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException

## steps

This was discovered while porting https://github.com/sbt/sbt-javaagent/ in https://github.com/sbt/sbt-javaagent/pull/48

```
scripted agent/confs
```

## problem

```
[info] [info] Defining outputStrategy
[info] [info] The new value will be used by Compile / forkOptions, Compile / run / forkOptions and 2 others.
[info] [info] 	Run `last` for details.
[info] [info] Reapplying settings...
[info] [info] set current project to agentConfs (in build file:/tmp/sbt_c062636d/)
[info] [info] compiling 1 Scala source to /tmp/sbt_c062636d/target/out/jvm/scala-3.7.2/agentconfs/test-backend ...
[info] [info] done compiling
Error:  Exception in thread "Simple-output-spawn-Thread-53" com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setStrictness(Strictness.LENIENT) to accept malformed JSON at line 1 column 8 path $
Error:  See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
Error:  	at com.google.gson.JsonParser.parseReader(JsonParser.java:116)
Error:  	at com.google.gson.JsonParser.parseString(JsonParser.java:92)
Error:  	at sbt.React.apply(ForkTests.scala:190)
Error:  	at sbt.React.apply(ForkTests.scala:188)
Error:  	at sbt.internal.WorkerExchange$.onStdoutLine$$anonfun$1(WorkerExchange.scala:67)
Error:  	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
Error:  	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
Error:  	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:619)
Error:  	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:617)
Error:  	at scala.collection.AbstractIterable.foreach(Iterable.scala:935)
Error:  	at sbt.internal.WorkerExchange$.onStdoutLine(WorkerExchange.scala:66)
Error:  	at sbt.internal.WorkerExchange$.$anonfun$2(WorkerExchange.scala:43)
Error:  	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
Error:  	at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
Error:  	at scala.sys.process.BasicIO$.$anonfun$processFully$1(BasicIO.scala:210)
Error:  	at scala.sys.process.BasicIO$.$anonfun$processFully$1$adapted(BasicIO.scala:188)
Error:  	at scala.sys.process.ProcessBuilderImpl$Simple.$anonfun$run$3(ProcessBuilderImpl.scala:91)
Error:  	at scala.sys.process.ProcessImpl$Spawn$$anon$1.run(ProcessImpl.scala:28)
Error:  Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setStrictness(Strictness.LENIENT) to accept malformed JSON at line 1 column 8 path $
Error:  See https://github.com/google/gson/blob/main/Troubleshooting.md#malformed-json
Error:  	at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1817)
Error:  	at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1613)
Error:  	at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:672)
Error:  	at com.google.gson.stream.JsonReader.peek(JsonReader.java:547)
Error:  	at com.google.gson.JsonParser.parseReader(JsonParser.java:111)
Error:  	... 17 more
[info] [info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] [info] No tests to run for Test / testQuick
[info] [success] elapsed time: 1 s, cache 66%, 28 disk cache hits, 14 onsite tasks
[info] [error] elapsed time: 0 s
Error:  x agent/confs 
Error:   Cause of test exception: {line 6}  Command failed: checkTestLog failed
```

## Notes

Not sure if this is a supported usage.

--- a/main-actions/src/main/scala/sbt/ForkTests.scala
+++ b/main-actions/src/main/scala/sbt/ForkTests.scala
@@ -8,7 +8,7 @@
 
 package sbt
 
-import com.google.gson.{ JsonObject, JsonParser }
+import com.google.gson.{ JsonObject, JsonParser, JsonSyntaxException }
 import testing.{ Logger as _, Task as _, * }
 import java.io.*
 import java.util.ArrayList
@@ -25,6 +25,7 @@ import scala.collection.mutable
 import scala.concurrent.{ Await, Promise }
 import scala.concurrent.duration.Duration
 import scala.util.Random
+import scala.util.control.NonFatal
 import scala.jdk.CollectionConverters.*
 import scala.sys.process.Process
 
@@ -186,17 +187,20 @@ private class React(
   val g = WorkerMain.mkGson()
   val promise: Promise[Int] = Promise()
   override def apply(line: String): Unit =
-    // scala.Console.err.println(line)
-    val o = JsonParser.parseString(line).getAsJsonObject()
-    if o.has("id") then
-      val resId = o.getAsJsonPrimitive("id").getAsLong()
-      if resId == id then
-        if promise.isCompleted then ()
-        else if o.has("error") then promise.failure(new RuntimeException(line))
-        else promise.success(0)
+    try
+      val o = JsonParser.parseString(line).getAsJsonObject()
+      if o.has("id") then
+        val resId = o.getAsJsonPrimitive("id").getAsLong()
+        if resId == id then
+          if promise.isCompleted then ()
+          else if o.has("error") then promise.failure(new RuntimeException(line))
+          else promise.success(0)
+        else ()
+      else if o.has("method") then processNotification(o)
       else ()
-    else if o.has("method") then processNotification(o)
-    else ()
+    catch
+      case _: JsonSyntaxException => log.info(line)
+      case NonFatal(_)            => ()
 
   override def notifyExit(p: Process): Unit =
     if !process.isAlive then promise.success(process.exitValue())

